lms/tasks/email_digests.py

```diff
@@ -40,35 +40,57 @@ def send_instructor_email_digest_tasks(batch_size):
 
     with app.request_context() as request:  # pylint:disable=no-member
         with request.tm:
-            h_userids = request.db.scalars(
-                select(User.h_userid)
-                .select_from(User)
-                .distinct()
-                .join(ApplicationInstance)
-                .join(AssignmentMembership)
-                .join(LTIRole)
-                .where(
-                    ApplicationInstance.settings["hypothesis"][
-                        "instructor_email_digests_enabled"
-                    ].astext
-                    == "true",
-                    LTIRole.type == "instructor",
-                )
-            ).all()
+            if not h_userids:
+                try:
+                    h_userids = request.db.scalars(
```

Comment:
So if the `h_userids` is given then it just uses it, otherwise it queries the DB to get the `h_userids` itself.

Normally `h_userids` won't be given so it'll query the DB. But when retrying only the `h_userids` that failed will be given